dvc/render/convert.py

```diff
@@ -0,0 +1,56 @@
+import json
+from typing import Dict, List, Union
+
+from dvc_render import ImageRenderer, VegaRenderer
+
+from dvc.render import REVISION_FIELD, REVISIONS_KEY, SRC_FIELD, TYPE_KEY
+from dvc.render.image_converter import ImageConverter
+from dvc.render.vega_converter import VegaConverter
+
+CONVERTERS = {
+    VegaRenderer.TYPE: VegaConverter,
+    ImageRenderer.TYPE: ImageConverter,
+}
+
+
+def to_datapoints(renderer_class, data: Dict, props: Dict):
+    converter: Union[VegaConverter, ImageConverter] = CONVERTERS[
+        renderer_class.TYPE
+    ](props)
+
+    datapoints = []
+    for revision, rev_data in data.items():
+        for filename, file_data in rev_data.get("data", {}).items():
+            if "data" in file_data:
+                processed, final_props = converter.convert(
+                    revision, filename, file_data.get("data")
+                )
+                datapoints.extend(processed)
+    return datapoints, final_props
+
+
+def to_json(renderer) -> List[Dict]:
+    if renderer.TYPE == "vega":
+        # TODO: Adapt to #7367
```

Comment:
@pared whichever format is decided in https://github.com/iterative/dvc/pull/7367 I think all the information could be extracted from a renderer instance so the conversion logic can be isolated here.